nexus/db-model/src/schema.rs

```diff
@@ -13,7 +13,7 @@ use omicron_common::api::external::SemverVersion;
 ///
 /// This should be updated whenever the schema is changed. For more details,
 /// refer to: schema/crdb/README.adoc
-pub const SCHEMA_VERSION: SemverVersion = SemverVersion::new(36, 0, 0);
+pub const SCHEMA_VERSION: SemverVersion = SemverVersion::new(37, 0, 1);
```

Comment:
Why do we have two schema versions in this PR?

Why do we use the micro number?  I thought schema versions were always supposed to bump the major.

Reply:
The reason we need two schema versions is that we have to make two separate transactions to achieve an idempotent result.

The order of operations we need to perform is:

1. Add the new `sled_policy` and `sled_state` enum types.
2. Add the new `sled_policy` and `sled_state` columns.
3. Update `sled_policy` to the new value based on the `provision_state` column.
4. Remove the `provision_state` column.
5. Remove the `sled_provision_state` enum types.

1 and 2 are inherently idempotent, so those are fine.

3 by itself is also fine to repeat.

But consider what happens if step 4 occurs, then the process fails, and we go back to 1 again. We'd repeat 3, but the corresponding column no longer exists so it will fail.

Solving this can be done in one of two ways:

1. Find a way to make 3 only perform the update if the `provision_state` column exists. This is possible, but [somewhat complicated](https://stackoverflow.com/questions/30131165/only-perform-update-if-column-exists).
2. Add a transaction commit between 3 and 4. This is the approach I ended up using.

The reason it's 37.0.0 and 37.0.1 is because both updates are part of the same PR. This follows the pattern in #4261.